#
# Various small fixes from CVS that are considered safe to apply to 2.8.6.
#
--- xchat-2.8.6/src/common/cfgfiles.c	2008-02-05 21:02:47.000000000 +1100
+++ src/common/cfgfiles.c	2008-06-15 13:45:43.000000000 +1000
@@ -886,7 +886,6 @@
 set_showval (session *sess, const struct prefs *var, char *tbuf)
 {
 	int len, dots, j;
-	static const char *offon[] = { "OFF", "ON" };
 
 	len = strlen (var->name);
 	memcpy (tbuf, var->name, len);
@@ -909,8 +908,10 @@
 					*((int *) &prefs + var->offset));
 		break;
 	case TYPE_BOOL:
-		sprintf (tbuf + len, "\0033:\017 %s\n", offon[
-					*((int *) &prefs + var->offset)]);
+		if (*((int *) &prefs + var->offset))
+			sprintf (tbuf + len, "\0033:\017 %s\n", "ON");
+		else
+			sprintf (tbuf + len, "\0033:\017 %s\n", "OFF");
 		break;
 	}
 	PrintText (sess, tbuf);
--- xchat-2.8.6/src/common/chanopt.c	2008-06-10 22:00:55.000000000 +1000
+++ src/common/chanopt.c	2008-06-15 13:48:04.000000000 +1000
@@ -32,7 +32,7 @@
 
 #define S_F(xx) STRUCT_OFFSET_STR(struct session,xx)
 
-channel_options chanopt[] =
+static const channel_options chanopt[] =
 {
 	{"alert_beep", "BEEP", S_F(alert_beep)},
 	{"alert_taskbar", NULL, S_F(alert_taskbar)},
--- xchat-2.8.6/src/common/servlist.c	2008-04-01 19:22:34.000000000 +1100
+++ src/common/servlist.c	2008-06-15 13:57:41.000000000 +1000
@@ -509,6 +509,8 @@
 	list = g_slist_nth (net->servlist, net->selected);
 	if (!list)
 		list = net->servlist;
+	if (!list)
+		return;
 	ircserv = list->data;
 
 	/* incase a protocol switch is added to the servlist gui */
--- xchat-2.8.6/src/common/text.c	2008-03-28 13:20:04.000000000 +1100
+++ src/common/text.c	2008-06-15 13:59:59.000000000 +1000
@@ -216,7 +216,7 @@
 static void
 scrollback_save (session *sess, char *text)
 {
-	char buf[1024];
+	char buf[512 * 4];
 	time_t stamp;
 	int len;
 
@@ -266,7 +266,7 @@
 scrollback_load (session *sess)
 {
 	int fh;
-	char buf[1024];
+	char buf[512 * 4];
 	char *text;
 	time_t stamp;
 	int lines;
